Packages/Core/Editor/Generator/Generator.cs

```diff
@@ -21,6 +22,191 @@ public static string[] GetTemplatePaths()
             return Directory.GetFiles(templateSearchPath, "UA_Template*.txt", SearchOption.AllDirectories);
         }
 
+        public static Dictionary<string, string> CreateTemplateVariablesMap(Type type, bool asPair, string withNamespace = default)
+        {
+            var typeName = type.FullName.Replace('+', '.'); // This accounts for nested types.
+            var displayNameNoPair = type.Name.Capitalize();
+            var displayName = displayNameNoPair;
+
+            if (asPair)
+            {
+                typeName = $"Pair<{typeName}>";
+                displayName += "Pair";
+            }
+
+            // Add Types.
+            var templateVariables = new Dictionary<string, string>()
+            {
+                { "VALUE_TYPE", typeName },
+                { "VALUE_TYPE_NAME", displayName },
+                { "VALUE_TYPE_NAME_NO_PAIR", displayNameNoPair }
+            };
+
+            // Add Namespaces if applicable.
+            var typeNamespace = type.Namespace;
+            if (!string.IsNullOrEmpty(typeNamespace))
+            {
+                templateVariables.Add("TYPE_NAMESPACE", typeNamespace);
+            }
+            if (!string.IsNullOrEmpty(withNamespace))
+            {
+                templateVariables.Add("SUB_UA_NAMESPACE", withNamespace);
+            }
+
+            return templateVariables;
+        }
+
+        public static List<string> CreateTemplateConditions(Type type, string withNamespace = default)
```

Comment:
Missing documentation for the method, the Generate method has good documentation, as an example.

Reply:
Same as above.